packages/web-app-files/tests/unit/components/SideBar/Links/PublicLinks/LinkEdit.spec.js

```diff
@@ -0,0 +1,542 @@
+            })
+            it('should enable link create button if form is valid', async () => {
+              const addLinkSpy = jest.spyOn(mapActions, 'addLink')
+              const wrapper = getMountedWrapper(
+                {
+                  linkInEdit: { name: 'Public Link', hasPassword: true },
+                  ...getLinkCapabilities({ enforcedExpireDate: true })
+                },
+                { saving: false }
+              )
+              const linkCreateButton = wrapper.find(selectors.linkCreateButton)
+              expect(linkCreateButton.attributes('disabled')).toBeFalsy()
+
+              await linkCreateButton.trigger('click')
+              expect(wrapper.vm.saving).toBeTruthy()
+              expect(addLinkSpy).toHaveBeenCalledTimes(1)
+            })
+          })
+          describe('when existing link is being updated', () => {
+            it('should show link save button', () => {
+              const wrapper = getShallowMountedWrapper(
+                {
+                  linkInEdit: { id: 1224, name: 'Public Link' }
+                },
+                { saving: false }
+              )
+
+              const linkSaveButton = wrapper.find(selectors.linkSaveButton)
+              expect(linkSaveButton.exists()).toBeTruthy()
+            })
+            it('should set link save button as disabled if form is not valid', () => {
+              const wrapper = getShallowMountedWrapper(
+                {
+                  linkInEdit: { id: 1224, name: 'Public Link' },
+                  // invalid password field
+                  // by default viewer role is selected with permission 1 (viewer)
+                  // combined with enforced for password is enforce for link form
+                  // since link password is enforced and password is not set, form will be invalid
+                  ...getLinkCapabilities({
+                    passwordEnforcedFor: {
+                      read_only: '1'
+                    }
+                  })
+                },
+                { saving: false }
+              )
+              const linkSaveButton = wrapper.find(selectors.linkSaveButton)
+
+              expect(linkSaveButton.attributes('disabled')).toBeTruthy()
+            })
+            it('should set link save button as disabled if form does not have any changes', () => {
+              const wrapper = getShallowMountedWrapper(
+                {
+                  linkInEdit: { id: 1224, name: 'Public Link' },
+                  ...getLinkCapabilities({
+                    passwordEnforcedFor: {
+                      read_only: '1'
+                    }
+                  })
+                },
+                { saving: false }
+              )
+              const linkSaveButton = wrapper.find(selectors.linkSaveButton)
+
+              expect(linkSaveButton.attributes('disabled')).toBeTruthy()
+            })
+
+            describe('when form is valid and has some changes', () => {
```

Comment:
do we also test if the form is invalid?

Reply:
i've added some tests to check if the form is valid